src/main/java/kitchenpos/menu/tobe/application/event/ProductPriceChangedListener.java

```diff
@@ -0,0 +1,36 @@
+package kitchenpos.menu.tobe.application.event;
+
+
+import kitchenpos.menu.tobe.domain.menu.Menu;
+import kitchenpos.menu.tobe.domain.menu.MenuRepository;
+import kitchenpos.common.event.publisher.ProductPriceChangedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Component
+public class ProductPriceChangedListener {
+    private final MenuRepository menuRepository;
+
+    public ProductPriceChangedListener(MenuRepository menuRepository) {
+        this.menuRepository = menuRepository;
+    }
+
+    @EventListener
+    @Transactional
+    public void handle(ProductPriceChangedEvent event) {
+        List<Menu> menus = menuRepository.findAllByProductId(event.productId());
+
+        for (Menu menu : menus) {
+            try {
+                menu.changePrice(event.price());
+            } catch (IllegalArgumentException e) {
+                menu.hide();
+            }
```

Comment:
이벤트 리스너가 비즈니스 로직을 수행하고있는 것 같네요. 🤔
아래와 같이 변경하고  menu 객체가 가격을 변경하는 것은 스스로 판단해서 진행하도록 수행하도록해보면 어떨까요? 🙂


```suggestion
            menu.changePrice(event.price());
```